src/pybind/pynmodl.cpp

```diff
@@ -164,6 +213,49 @@ PYBIND11_MODULE(_nmodl, m_nmodl) {
+        .def_readwrite("sympy_cse", &nmodl::codegen::CodeGenConfig::sympy_cse)
+        .def_readwrite("sympy_conductance", &nmodl::codegen::CodeGenConfig::sympy_conductance)
+        .def_readwrite("nmodl_inline", &nmodl::codegen::CodeGenConfig::nmodl_inline)
+        .def_readwrite("nmodl_unroll", &nmodl::codegen::CodeGenConfig::nmodl_unroll)
+        .def_readwrite("nmodl_const_folding", &nmodl::codegen::CodeGenConfig::nmodl_const_folding)
+        .def_readwrite("nmodl_localize", &nmodl::codegen::CodeGenConfig::nmodl_localize)
+        .def_readwrite("nmodl_global_to_range", &nmodl::codegen::CodeGenConfig::nmodl_global_to_range)
+        .def_readwrite("nmodl_local_to_range", &nmodl::codegen::CodeGenConfig::nmodl_local_to_range)
+        .def_readwrite("localize_verbatim", &nmodl::codegen::CodeGenConfig::localize_verbatim)
+        .def_readwrite("local_rename", &nmodl::codegen::CodeGenConfig::local_rename)
+        .def_readwrite("verbatim_inline", &nmodl::codegen::CodeGenConfig::verbatim_inline)
+        .def_readwrite("verbatim_rename", &nmodl::codegen::CodeGenConfig::verbatim_rename)
+        .def_readwrite("force_codegen", &nmodl::codegen::CodeGenConfig::force_codegen)
+        .def_readwrite("only_check_compatibility", &nmodl::codegen::CodeGenConfig::only_check_compatibility)
+        .def_readwrite("optimize_ionvar_copies_codegen", &nmodl::codegen::CodeGenConfig::optimize_ionvar_copies_codegen)
+        .def_readwrite("output_dir", &nmodl::codegen::CodeGenConfig::output_dir)
+        .def_readwrite("scratch_dir", &nmodl::codegen::CodeGenConfig::scratch_dir)
+        .def_readwrite("data_type", &nmodl::codegen::CodeGenConfig::data_type)
+        .def_readwrite("llvm_ir", &nmodl::codegen::CodeGenConfig::llvm_ir)
+        .def_readwrite("llvm_float_type", &nmodl::codegen::CodeGenConfig::llvm_float_type)
+        .def_readwrite("llvm_opt_level_ir", &nmodl::codegen::CodeGenConfig::llvm_opt_level_ir)
+        .def_readwrite("llvm_math_library", &nmodl::codegen::CodeGenConfig::llvm_math_library)
+        .def_readwrite("llvm_no_debug", &nmodl::codegen::CodeGenConfig::llvm_no_debug)
+        .def_readwrite("llvm_fast_math_flags", &nmodl::codegen::CodeGenConfig::llvm_fast_math_flags)
+        .def_readwrite("llvm_cpu_name", &nmodl::codegen::CodeGenConfig::llvm_cpu_name)
+        .def_readwrite("llvm_gpu_name", &nmodl::codegen::CodeGenConfig::llvm_gpu_name)
+        .def_readwrite("llvm_vector_width", &nmodl::codegen::CodeGenConfig::llvm_vector_width)
+        .def_readwrite("llvm_opt_level_codegen", &nmodl::codegen::CodeGenConfig::llvm_opt_level_codegen)
+        .def_readwrite("shared_lib_paths", &nmodl::codegen::CodeGenConfig::shared_lib_paths);
+
+    py::class_<nmodl::JitDriver> jit_driver(m_nmodl, "Jit", nmodl::docstring::jit);
+    jit_driver.def(py::init<>())
+        .def("run", &nmodl::JitDriver::run, "node"_a, "modname"_a, "num_experiments"_a, "instance_size"_a);
```

Comment:
Not sure how useful this interface is in its current form to a user. But I agree with you that ultimately we should have some reasonable defaults set. For our benchmarks it would be best to stay explicit.